community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPage.java

```diff
@@ -47,6 +50,10 @@
 
     private long pointer;
 
+    // max transaction id that updated this page
+    @SuppressWarnings( "unused" )
+    private volatile long lastModifiedTxId = UNBOUND_LAST_MODIFIED_TX_ID;
```

Comment:
Since you're changing this field in some cases with `Unsafe`, could you take that all the way and always change/read it with `Unsafe` and remove `volatile`?

Reply:
I like keeping the `volatile` modifier as a reminder that this is the mode of access used on this field.

Reply:
i would like (and actually did that), to keep it as volatile for exactly the same reason :)